interface.go

```diff
@@ -30,3 +30,41 @@ type Logger interface {
 	Println(...interface{})
 	Printf(string, ...interface{})
 }
+
+// StreamErrorHandlerResult is a constant type for use with StreamErrorHandler.
+type StreamErrorHandlerResult string
```

Comment:
Maybe using these constants is overkill— there are only two of them, so it could just be a bool. My rationale was 1. enum-like semantics make the code easier to understand (does "true" mean proceed or stop), and 2. maybe there is some other kind of error handler result we might want to return in the future, though I can't think of any right now. On the other hand, enum-like behavior in Go is kind of problematic because you can't enforce that it is really one of those two values.

Two alternatives:

1. Just have the function return bool. They have to read the doc comment to know what that means.
2. Have this type be a struct, with a single field: `StopStream bool`. That way if we do ever need to add something, we don't need to change the function signature; also, the meaning is clearer.